### PR TITLE
Socket IO integrated, some room stuff working

### PR DESCRIPTION
**Pull request by jonahryandavis**
Added missing client files

Server to room communication implemented

Fixed merge issues

@CecileRobertMichon Im guessing you'll need to change a bunch of things about room.html and how client.js populates it

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -66,6 +64,33 @@ app.post('/newgame', function(request, response){
...
+            io.to(gameid).emit('initRoom', room.toString())
+        }
+        else {
+            io.to(gameid).emit('roomDeleted')
+        }
+    });
+
+    // Handle an example event with example data
+    socket.on('exampleClientEvent', function(exampleData) {
+        for(var room in socket.rooms) { // Loop through every room this connection belongs to
```
Comment (jonahryandavis) [inline index.js]: I think there might be a better way to do this but for now if a connection belongs to a room, then any change they send the server will be treated as if it affects that room

**Comment by j-mendonca** [comment]
Nice, this is awesome!

**Comment by jkirman** [comment]
Looks good, we'll integrate this with the player logic we've been working on.

**Comment by jonahryandavis** [comment]
@jkirman I just updated this a little more with very basic player logic for testing. You probably have to get rid of my stuff I just wanted to see if rooms work with multiple people

**Pull request merged**